// shippo/models/operations/deleteuserparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteUserParcelTemplateRequest$ = exports.DeleteUserParcelTemplateRequest$outboundSchema = exports.DeleteUserParcelTemplateRequest$inboundSchema = exports.DeleteUserParcelTemplateGlobals$ = exports.DeleteUserParcelTemplateGlobals$outboundSchema = exports.DeleteUserParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.DeleteUserParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.DeleteUserParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteUserParcelTemplateGlobals$;
(function (DeleteUserParcelTemplateGlobals$) {
    /** @deprecated use `DeleteUserParcelTemplateGlobals$inboundSchema` instead. */
    DeleteUserParcelTemplateGlobals$.inboundSchema = exports.DeleteUserParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `DeleteUserParcelTemplateGlobals$outboundSchema` instead. */
    DeleteUserParcelTemplateGlobals$.outboundSchema = exports.DeleteUserParcelTemplateGlobals$outboundSchema;
})(DeleteUserParcelTemplateGlobals$ || (exports.DeleteUserParcelTemplateGlobals$ = DeleteUserParcelTemplateGlobals$ = {}));
/** @internal */
exports.DeleteUserParcelTemplateRequest$inboundSchema = z.object({
    UserParcelTemplateObjectId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "UserParcelTemplateObjectId": "userParcelTemplateObjectId",
    });
});
/** @internal */
exports.DeleteUserParcelTemplateRequest$outboundSchema = z.object({
    userParcelTemplateObjectId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userParcelTemplateObjectId: "UserParcelTemplateObjectId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DeleteUserParcelTemplateRequest$;
(function (DeleteUserParcelTemplateRequest$) {
    /** @deprecated use `DeleteUserParcelTemplateRequest$inboundSchema` instead. */
    DeleteUserParcelTemplateRequest$.inboundSchema = exports.DeleteUserParcelTemplateRequest$inboundSchema;
    /** @deprecated use `DeleteUserParcelTemplateRequest$outboundSchema` instead. */
    DeleteUserParcelTemplateRequest$.outboundSchema = exports.DeleteUserParcelTemplateRequest$outboundSchema;
})(DeleteUserParcelTemplateRequest$ || (exports.DeleteUserParcelTemplateRequest$ = DeleteUserParcelTemplateRequest$ = {}));
