// shippo/models/operations/getmanifest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetManifestRequest$ = exports.GetManifestRequest$outboundSchema = exports.GetManifestRequest$inboundSchema = exports.GetManifestGlobals$ = exports.GetManifestGlobals$outboundSchema = exports.GetManifestGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetManifestGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetManifestGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetManifestGlobals$;
(function (GetManifestGlobals$) {
    /** @deprecated use `GetManifestGlobals$inboundSchema` instead. */
    GetManifestGlobals$.inboundSchema = exports.GetManifestGlobals$inboundSchema;
    /** @deprecated use `GetManifestGlobals$outboundSchema` instead. */
    GetManifestGlobals$.outboundSchema = exports.GetManifestGlobals$outboundSchema;
})(GetManifestGlobals$ || (exports.GetManifestGlobals$ = GetManifestGlobals$ = {}));
/** @internal */
exports.GetManifestRequest$inboundSchema = z.object({
    ManifestId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "ManifestId": "manifestId",
    });
});
/** @internal */
exports.GetManifestRequest$outboundSchema = z.object({
    manifestId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        manifestId: "ManifestId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetManifestRequest$;
(function (GetManifestRequest$) {
    /** @deprecated use `GetManifestRequest$inboundSchema` instead. */
    GetManifestRequest$.inboundSchema = exports.GetManifestRequest$inboundSchema;
    /** @deprecated use `GetManifestRequest$outboundSchema` instead. */
    GetManifestRequest$.outboundSchema = exports.GetManifestRequest$outboundSchema;
})(GetManifestRequest$ || (exports.GetManifestRequest$ = GetManifestRequest$ = {}));
