// shippo/models/operations/getuserparceltemplate.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetUserParcelTemplateRequest$ = exports.GetUserParcelTemplateRequest$outboundSchema = exports.GetUserParcelTemplateRequest$inboundSchema = exports.GetUserParcelTemplateGlobals$ = exports.GetUserParcelTemplateGlobals$outboundSchema = exports.GetUserParcelTemplateGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
/** @internal */
exports.GetUserParcelTemplateGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.GetUserParcelTemplateGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetUserParcelTemplateGlobals$;
(function (GetUserParcelTemplateGlobals$) {
    /** @deprecated use `GetUserParcelTemplateGlobals$inboundSchema` instead. */
    GetUserParcelTemplateGlobals$.inboundSchema = exports.GetUserParcelTemplateGlobals$inboundSchema;
    /** @deprecated use `GetUserParcelTemplateGlobals$outboundSchema` instead. */
    GetUserParcelTemplateGlobals$.outboundSchema = exports.GetUserParcelTemplateGlobals$outboundSchema;
})(GetUserParcelTemplateGlobals$ || (exports.GetUserParcelTemplateGlobals$ = GetUserParcelTemplateGlobals$ = {}));
/** @internal */
exports.GetUserParcelTemplateRequest$inboundSchema = z.object({
    UserParcelTemplateObjectId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "UserParcelTemplateObjectId": "userParcelTemplateObjectId",
    });
});
/** @internal */
exports.GetUserParcelTemplateRequest$outboundSchema = z.object({
    userParcelTemplateObjectId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        userParcelTemplateObjectId: "UserParcelTemplateObjectId",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var GetUserParcelTemplateRequest$;
(function (GetUserParcelTemplateRequest$) {
    /** @deprecated use `GetUserParcelTemplateRequest$inboundSchema` instead. */
    GetUserParcelTemplateRequest$.inboundSchema = exports.GetUserParcelTemplateRequest$inboundSchema;
    /** @deprecated use `GetUserParcelTemplateRequest$outboundSchema` instead. */
    GetUserParcelTemplateRequest$.outboundSchema = exports.GetUserParcelTemplateRequest$outboundSchema;
})(GetUserParcelTemplateRequest$ || (exports.GetUserParcelTemplateRequest$ = GetUserParcelTemplateRequest$ = {}));
