// shippo/models/operations/updatecarrieraccount.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCarrierAccountRequest$ = exports.UpdateCarrierAccountRequest$outboundSchema = exports.UpdateCarrierAccountRequest$inboundSchema = exports.UpdateCarrierAccountGlobals$ = exports.UpdateCarrierAccountGlobals$outboundSchema = exports.UpdateCarrierAccountGlobals$inboundSchema = void 0;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.UpdateCarrierAccountGlobals$inboundSchema = z.object({
    "SHIPPO-API-VERSION": z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "SHIPPO-API-VERSION": "shippoApiVersion",
    });
});
/** @internal */
exports.UpdateCarrierAccountGlobals$outboundSchema = z.object({
    shippoApiVersion: z.string().optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        shippoApiVersion: "SHIPPO-API-VERSION",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateCarrierAccountGlobals$;
(function (UpdateCarrierAccountGlobals$) {
    /** @deprecated use `UpdateCarrierAccountGlobals$inboundSchema` instead. */
    UpdateCarrierAccountGlobals$.inboundSchema = exports.UpdateCarrierAccountGlobals$inboundSchema;
    /** @deprecated use `UpdateCarrierAccountGlobals$outboundSchema` instead. */
    UpdateCarrierAccountGlobals$.outboundSchema = exports.UpdateCarrierAccountGlobals$outboundSchema;
})(UpdateCarrierAccountGlobals$ || (exports.UpdateCarrierAccountGlobals$ = UpdateCarrierAccountGlobals$ = {}));
/** @internal */
exports.UpdateCarrierAccountRequest$inboundSchema = z.object({
    CarrierAccountId: z.string(),
    CarrierAccountBase: components.CarrierAccountBase$inboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "CarrierAccountId": "carrierAccountId",
        "CarrierAccountBase": "carrierAccountBase",
    });
});
/** @internal */
exports.UpdateCarrierAccountRequest$outboundSchema = z.object({
    carrierAccountId: z.string(),
    carrierAccountBase: components.CarrierAccountBase$outboundSchema.optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        carrierAccountId: "CarrierAccountId",
        carrierAccountBase: "CarrierAccountBase",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var UpdateCarrierAccountRequest$;
(function (UpdateCarrierAccountRequest$) {
    /** @deprecated use `UpdateCarrierAccountRequest$inboundSchema` instead. */
    UpdateCarrierAccountRequest$.inboundSchema = exports.UpdateCarrierAccountRequest$inboundSchema;
    /** @deprecated use `UpdateCarrierAccountRequest$outboundSchema` instead. */
    UpdateCarrierAccountRequest$.outboundSchema = exports.UpdateCarrierAccountRequest$outboundSchema;
})(UpdateCarrierAccountRequest$ || (exports.UpdateCarrierAccountRequest$ = UpdateCarrierAccountRequest$ = {}));
