// shippo/models/components/carrieraccountdpddecreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountDpdDeCreateRequest$ = exports.CarrierAccountDpdDeCreateRequest$outboundSchema = exports.CarrierAccountDpdDeCreateRequest$inboundSchema = exports.CarrierAccountDpdDeCreateRequestParameters$ = exports.CarrierAccountDpdDeCreateRequestParameters$outboundSchema = exports.CarrierAccountDpdDeCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CarrierAccountDpdDeCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountDpdDeCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDpdDeCreateRequestParameters$;
(function (CarrierAccountDpdDeCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountDpdDeCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountDpdDeCreateRequestParameters$.inboundSchema = exports.CarrierAccountDpdDeCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountDpdDeCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountDpdDeCreateRequestParameters$.outboundSchema = exports.CarrierAccountDpdDeCreateRequestParameters$outboundSchema;
})(CarrierAccountDpdDeCreateRequestParameters$ || (exports.CarrierAccountDpdDeCreateRequestParameters$ = CarrierAccountDpdDeCreateRequestParameters$ = {}));
/** @internal */
exports.CarrierAccountDpdDeCreateRequest$inboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountDpdDeCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountDpdDeCreateRequest$outboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountDpdDeCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDpdDeCreateRequest$;
(function (CarrierAccountDpdDeCreateRequest$) {
    /** @deprecated use `CarrierAccountDpdDeCreateRequest$inboundSchema` instead. */
    CarrierAccountDpdDeCreateRequest$.inboundSchema = exports.CarrierAccountDpdDeCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountDpdDeCreateRequest$outboundSchema` instead. */
    CarrierAccountDpdDeCreateRequest$.outboundSchema = exports.CarrierAccountDpdDeCreateRequest$outboundSchema;
})(CarrierAccountDpdDeCreateRequest$ || (exports.CarrierAccountDpdDeCreateRequest$ = CarrierAccountDpdDeCreateRequest$ = {}));
