// shippo/models/components/carrieraccountdpdukcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountDPDUKCreateRequest$ = exports.CarrierAccountDPDUKCreateRequest$outboundSchema = exports.CarrierAccountDPDUKCreateRequest$inboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$ = exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDPDUKCreateRequestParameters$;
(function (CarrierAccountDPDUKCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountDPDUKCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountDPDUKCreateRequestParameters$.inboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountDPDUKCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountDPDUKCreateRequestParameters$.outboundSchema = exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema;
})(CarrierAccountDPDUKCreateRequestParameters$ || (exports.CarrierAccountDPDUKCreateRequestParameters$ = CarrierAccountDPDUKCreateRequestParameters$ = {}));
/** @internal */
exports.CarrierAccountDPDUKCreateRequest$inboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountDPDUKCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountDPDUKCreateRequest$outboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountDPDUKCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountDPDUKCreateRequest$;
(function (CarrierAccountDPDUKCreateRequest$) {
    /** @deprecated use `CarrierAccountDPDUKCreateRequest$inboundSchema` instead. */
    CarrierAccountDPDUKCreateRequest$.inboundSchema = exports.CarrierAccountDPDUKCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountDPDUKCreateRequest$outboundSchema` instead. */
    CarrierAccountDPDUKCreateRequest$.outboundSchema = exports.CarrierAccountDPDUKCreateRequest$outboundSchema;
})(CarrierAccountDPDUKCreateRequest$ || (exports.CarrierAccountDPDUKCreateRequest$ = CarrierAccountDPDUKCreateRequest$ = {}));
