// shippo/models/components/carrieraccountuspscreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountUSPSCreateRequest$ = exports.CarrierAccountUSPSCreateRequest$outboundSchema = exports.CarrierAccountUSPSCreateRequest$inboundSchema = exports.CarrierAccountUSPSCreateRequestParameters$ = exports.CarrierAccountUSPSCreateRequestParameters$outboundSchema = exports.CarrierAccountUSPSCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CarrierAccountUSPSCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountUSPSCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountUSPSCreateRequestParameters$;
(function (CarrierAccountUSPSCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountUSPSCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountUSPSCreateRequestParameters$.inboundSchema = exports.CarrierAccountUSPSCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountUSPSCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountUSPSCreateRequestParameters$.outboundSchema = exports.CarrierAccountUSPSCreateRequestParameters$outboundSchema;
})(CarrierAccountUSPSCreateRequestParameters$ || (exports.CarrierAccountUSPSCreateRequestParameters$ = CarrierAccountUSPSCreateRequestParameters$ = {}));
/** @internal */
exports.CarrierAccountUSPSCreateRequest$inboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountUSPSCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountUSPSCreateRequest$outboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountUSPSCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountUSPSCreateRequest$;
(function (CarrierAccountUSPSCreateRequest$) {
    /** @deprecated use `CarrierAccountUSPSCreateRequest$inboundSchema` instead. */
    CarrierAccountUSPSCreateRequest$.inboundSchema = exports.CarrierAccountUSPSCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountUSPSCreateRequest$outboundSchema` instead. */
    CarrierAccountUSPSCreateRequest$.outboundSchema = exports.CarrierAccountUSPSCreateRequest$outboundSchema;
})(CarrierAccountUSPSCreateRequest$ || (exports.CarrierAccountUSPSCreateRequest$ = CarrierAccountUSPSCreateRequest$ = {}));
