// shippo/models/components/carrieraccounthermesukcreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountHermesUKCreateRequest$ = exports.CarrierAccountHermesUKCreateRequest$outboundSchema = exports.CarrierAccountHermesUKCreateRequest$inboundSchema = exports.CarrierAccountHermesUKCreateRequestParameters$ = exports.CarrierAccountHermesUKCreateRequestParameters$outboundSchema = exports.CarrierAccountHermesUKCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CarrierAccountHermesUKCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountHermesUKCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountHermesUKCreateRequestParameters$;
(function (CarrierAccountHermesUKCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountHermesUKCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountHermesUKCreateRequestParameters$.inboundSchema = exports.CarrierAccountHermesUKCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountHermesUKCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountHermesUKCreateRequestParameters$.outboundSchema = exports.CarrierAccountHermesUKCreateRequestParameters$outboundSchema;
})(CarrierAccountHermesUKCreateRequestParameters$ || (exports.CarrierAccountHermesUKCreateRequestParameters$ = CarrierAccountHermesUKCreateRequestParameters$ = {}));
/** @internal */
exports.CarrierAccountHermesUKCreateRequest$inboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountHermesUKCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountHermesUKCreateRequest$outboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountHermesUKCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountHermesUKCreateRequest$;
(function (CarrierAccountHermesUKCreateRequest$) {
    /** @deprecated use `CarrierAccountHermesUKCreateRequest$inboundSchema` instead. */
    CarrierAccountHermesUKCreateRequest$.inboundSchema = exports.CarrierAccountHermesUKCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountHermesUKCreateRequest$outboundSchema` instead. */
    CarrierAccountHermesUKCreateRequest$.outboundSchema = exports.CarrierAccountHermesUKCreateRequest$outboundSchema;
})(CarrierAccountHermesUKCreateRequest$ || (exports.CarrierAccountHermesUKCreateRequest$ = CarrierAccountHermesUKCreateRequest$ = {}));
