// shippo/models/components/carrieraccountcolissimocreaterequest.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CarrierAccountColissimoCreateRequest$ = exports.CarrierAccountColissimoCreateRequest$outboundSchema = exports.CarrierAccountColissimoCreateRequest$inboundSchema = exports.CarrierAccountColissimoCreateRequestParameters$ = exports.CarrierAccountColissimoCreateRequestParameters$outboundSchema = exports.CarrierAccountColissimoCreateRequestParameters$inboundSchema = void 0;
const z = __importStar(require("zod"));
/** @internal */
exports.CarrierAccountColissimoCreateRequestParameters$inboundSchema = z.object({});
/** @internal */
exports.CarrierAccountColissimoCreateRequestParameters$outboundSchema = z.object({});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountColissimoCreateRequestParameters$;
(function (CarrierAccountColissimoCreateRequestParameters$) {
    /** @deprecated use `CarrierAccountColissimoCreateRequestParameters$inboundSchema` instead. */
    CarrierAccountColissimoCreateRequestParameters$.inboundSchema = exports.CarrierAccountColissimoCreateRequestParameters$inboundSchema;
    /** @deprecated use `CarrierAccountColissimoCreateRequestParameters$outboundSchema` instead. */
    CarrierAccountColissimoCreateRequestParameters$.outboundSchema = exports.CarrierAccountColissimoCreateRequestParameters$outboundSchema;
})(CarrierAccountColissimoCreateRequestParameters$ || (exports.CarrierAccountColissimoCreateRequestParameters$ = CarrierAccountColissimoCreateRequestParameters$ = {}));
/** @internal */
exports.CarrierAccountColissimoCreateRequest$inboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountColissimoCreateRequestParameters$inboundSchema),
});
/** @internal */
exports.CarrierAccountColissimoCreateRequest$outboundSchema = z.object({
    carrier: z.string(),
    parameters: z.lazy(() => exports.CarrierAccountColissimoCreateRequestParameters$outboundSchema),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CarrierAccountColissimoCreateRequest$;
(function (CarrierAccountColissimoCreateRequest$) {
    /** @deprecated use `CarrierAccountColissimoCreateRequest$inboundSchema` instead. */
    CarrierAccountColissimoCreateRequest$.inboundSchema = exports.CarrierAccountColissimoCreateRequest$inboundSchema;
    /** @deprecated use `CarrierAccountColissimoCreateRequest$outboundSchema` instead. */
    CarrierAccountColissimoCreateRequest$.outboundSchema = exports.CarrierAccountColissimoCreateRequest$outboundSchema;
})(CarrierAccountColissimoCreateRequest$ || (exports.CarrierAccountColissimoCreateRequest$ = CarrierAccountColissimoCreateRequest$ = {}));
