// shippo/models/components/parceltemplatecourierspleaseenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ParcelTemplateCouriersPleaseEnum$ = exports.ParcelTemplateCouriersPleaseEnum$outboundSchema = exports.ParcelTemplateCouriersPleaseEnum$inboundSchema = exports.ParcelTemplateCouriersPleaseEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Name | Dimensions|
 *
 * @remarks
 * |:---|:---|:---|
 * | couriersplease_500g_satchel | 500g Satchel | 22.00 x 33.50 x 0.10 cm|
 * | couriersplease_1kg_satchel | 1kg Satchel | 28.00 x 35.00 x 0.10 cm|
 * | couriersplease_3kg_satchel | 3kg Satchel | 34.00 x 42.00 x 0.10 cm|
 * | couriersplease_5kg_satchel | 5kg Satchel | 43.70 x 59.70 x 0.10 cm|
 */
exports.ParcelTemplateCouriersPleaseEnum = {
    Couriersplease500gSatchel: "couriersplease_500g_satchel",
    Couriersplease1kgSatchel: "couriersplease_1kg_satchel",
    Couriersplease3kgSatchel: "couriersplease_3kg_satchel",
    Couriersplease5kgSatchel: "couriersplease_5kg_satchel",
};
/** @internal */
exports.ParcelTemplateCouriersPleaseEnum$inboundSchema = z.nativeEnum(exports.ParcelTemplateCouriersPleaseEnum);
/** @internal */
exports.ParcelTemplateCouriersPleaseEnum$outboundSchema = exports.ParcelTemplateCouriersPleaseEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ParcelTemplateCouriersPleaseEnum$;
(function (ParcelTemplateCouriersPleaseEnum$) {
    /** @deprecated use `ParcelTemplateCouriersPleaseEnum$inboundSchema` instead. */
    ParcelTemplateCouriersPleaseEnum$.inboundSchema = exports.ParcelTemplateCouriersPleaseEnum$inboundSchema;
    /** @deprecated use `ParcelTemplateCouriersPleaseEnum$outboundSchema` instead. */
    ParcelTemplateCouriersPleaseEnum$.outboundSchema = exports.ParcelTemplateCouriersPleaseEnum$outboundSchema;
})(ParcelTemplateCouriersPleaseEnum$ || (exports.ParcelTemplateCouriersPleaseEnum$ = ParcelTemplateCouriersPleaseEnum$ = {}));
