// shippo/models/components/servicelevelcourierspleaseenum.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceLevelCouriersPleaseEnum$ = exports.ServiceLevelCouriersPleaseEnum$outboundSchema = exports.ServiceLevelCouriersPleaseEnum$inboundSchema = exports.ServiceLevelCouriersPleaseEnum = void 0;
const z = __importStar(require("zod"));
/**
 * |Token | Service name|
 *
 * @remarks
 * |:---|:---|
 * | couriersplease_domestic_priority_auth_to_leave | Domestic Priority - Authority To Leave/POPPoints|
 * | couriersplease_domestic_priority_sign_required | Domestic Priority - Signature Required|
 * | couriersplease_gold_domestic_auth_to_leave | Gold Domestic - Authority To Leave/POPPoints|
 * | couriersplease_gold_domestic_sign_required | Gold Domestic - Signature Required|
 * | couriersplease_off_peak_auth_to_leave | Off Peak - Authority To Leave/POPPoints|
 * | couriersplease_off_peak_sign_required | Off Peak - Signature Required|
 * | couriersplease_parcel_auth_to_leave | Parcel - Authority To Leave|
 * | couriersplease_parcel_sign_required | Parcel - Signature Required|
 * | couriersplease_road_express | Road Express|
 * | couriersplease_satchel_auth_to_leave | Satchel - Authority To Leave|
 * | couriersplease_satchel_sign_required | Satchel - Signature Required|
 */
exports.ServiceLevelCouriersPleaseEnum = {
    CourierspleaseDomesticPriorityAuthToLeave: "couriersplease_domestic_priority_auth_to_leave",
    CourierspleaseDomesticPrioritySignRequired: "couriersplease_domestic_priority_sign_required",
    CourierspleaseGoldDomesticAuthToLeave: "couriersplease_gold_domestic_auth_to_leave",
    CourierspleaseGoldDomesticSignRequired: "couriersplease_gold_domestic_sign_required",
    CourierspleaseOffPeakAuthToLeave: "couriersplease_off_peak_auth_to_leave",
    CourierspleaseOffPeakSignRequired: "couriersplease_off_peak_sign_required",
    CourierspleaseParcelAuthToLeave: "couriersplease_parcel_auth_to_leave",
    CourierspleaseParcelSignRequired: "couriersplease_parcel_sign_required",
    CourierspleaseRoadExpress: "couriersplease_road_express",
    CourierspleaseSatchelAuthToLeave: "couriersplease_satchel_auth_to_leave",
    CourierspleaseSatchelSignRequired: "couriersplease_satchel_sign_required",
};
/** @internal */
exports.ServiceLevelCouriersPleaseEnum$inboundSchema = z.nativeEnum(exports.ServiceLevelCouriersPleaseEnum);
/** @internal */
exports.ServiceLevelCouriersPleaseEnum$outboundSchema = exports.ServiceLevelCouriersPleaseEnum$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ServiceLevelCouriersPleaseEnum$;
(function (ServiceLevelCouriersPleaseEnum$) {
    /** @deprecated use `ServiceLevelCouriersPleaseEnum$inboundSchema` instead. */
    ServiceLevelCouriersPleaseEnum$.inboundSchema = exports.ServiceLevelCouriersPleaseEnum$inboundSchema;
    /** @deprecated use `ServiceLevelCouriersPleaseEnum$outboundSchema` instead. */
    ServiceLevelCouriersPleaseEnum$.outboundSchema = exports.ServiceLevelCouriersPleaseEnum$outboundSchema;
})(ServiceLevelCouriersPleaseEnum$ || (exports.ServiceLevelCouriersPleaseEnum$ = ServiceLevelCouriersPleaseEnum$ = {}));
